apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexService.java

```diff
@@ -139,5 +147,27 @@ public boolean shouldFreezeAPL(int offerType, int dexCurrencies){
         return false;
     }
 
+    /**
+     * @param cancelTrId  can be null if we just want to check are there any unconfirmed transactions for this order.
+     */
+    public boolean isThereAnotherCancelUnconfirmedTr(Long orderId, Long cancelTrId){
+        try(DbIterator<UnconfirmedTransaction>  tr = transactionProcessor.getAllUnconfirmedTransactions()) {
+            while (tr.hasNext()) {
+                UnconfirmedTransaction unconfirmedTransaction = tr.next();
+                if (TransactionType.TYPE_DEX == unconfirmedTransaction.getTransaction().getType().getType() &&
```

Comment:
I would add more NPE checks here for more safety:

```
if ( unconfirmedTransaction != null &&  unconfirmedTransaction.getTransaction() != null
&&  unconfirmedTransaction.getTransaction().getType() != null &&  unconfirmedTransaction.getTransaction().getType().getType() != null.............. )
```

Reply:
tx.next() coudn't return null.
unconfirmedTransaction can't be without transaction.
Transaction can't be without Type.
.getType().getType() is a primitive (byte)